frontend/src/forms/MonsterForm.js

```diff
@@ -0,0 +1,53 @@
+import React, { useContext, useState } from 'react';
+import MonsterContext from '../contexts/MonsterContext';
+import { useHistory } from 'react-router-dom';
+import MonsterGallery from './MonsterGallery';
+
+const initialState = {
+  name: '',
+};
+
+export default function MonsterForm({ monster = initialState }) {
+  const [monsterData, setMonsterData] = useState(monster);
+  const { create } = useContext(MonsterContext);
+  const history = useHistory();
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <MonsterGallery />
+      <label>
+        name
+        <input
+          name="name"
+          value={monsterData.name}
+          onChange={handleChange}
+          type="text"
+          required
+        />
+      </label>
+      <button id="save">Save</button>
+      <button type="button" onClick={handleCancel}>
+        Cancel
+      </button>
+    </form>
+  );
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    handleSave(monsterData.name);
+  }
+
+  function handleSave() {
+    const monsterImage = window.document.getElementById('currentImage').src;
```

Comment:
This is really bad as it couples this component via the id attribute and accesses data from the dom directly. Find a way to store your image in a state and make react the single source of truth.